internal/services/synapse/synapse_firewall_rule_resource_test.go

```diff
@@ -78,7 +78,7 @@ func (r SynapseFirewallRuleResource) Exists(ctx context.Context, client *clients
 		if utils.ResponseWasNotFound(resp.Response) {
 			return utils.Bool(false), nil
 		}
-		return nil, fmt.Errorf("retrieving Synapse Firewall Rule %q (Workspace %q / Resource Group %q): %+v", id.Name, id.WorkspaceName, id.ResourceGroup, err)
+		return nil, fmt.Errorf("retrieving %s: %+v", id, err)
```

Comment:
this is a parsed ID, so it'll be returned as a pointer - as such this needs to be:

```suggestion
		return nil, fmt.Errorf("retrieving %s: %+v", *id, err)
```